### PR TITLE
Dependancy resolution

### PR DESCRIPTION
**Pull request by Khitiara**

### CODE REVIEW

File: gui/dialogs/ChooseInstallModDialog.cpp [hunk 1]
```diff
@@ -9,166 +9,189 @@
...
+            tags.append(QString("<a href=\"%1\">%1</a>").arg(tag));
+        }
+        ui->tagsLabel->setText(tags.join(", "));
+        ui->logoLabel->setPixmap(mod->logo());
+    }
 }
 
 #include "ChooseInstallModDialog.moc"
+
+void ChooseInstallModDialog::on_pushButton_clicked()
```
Comment (02JanDal) [inline gui/dialogs/ChooseInstallModDialog.cpp]: Please rename pushButton and pushButton_2 to something more describing

File: gui/dialogs/ChooseInstallModDialog.cpp [hunk 2]
```diff
@@ -9,166 +9,189 @@
...
+
+    auto choosenMod = m_view->selectionModel()->selectedRows().first().data(QuickModsList::QuickModRole).value<QuickMod*>();
+    QList<QuickMod*> modsList;
+
+    modsList.append(choosenMod);
+    DownloadProgressDialog dialog(choosenMod->dependentUrls().count(), this);
+    connect(&*MMC->quickmodslist(), SIGNAL(modAdded(QuickMod*)), &dialog, SLOT(ModAdded(QuickMod*)));
+    for(auto url : choosenMod->dependentUrls())
+    {
+        MMC->quickmodslist()->registerMod(url);
```
Comment (02JanDal) [inline gui/dialogs/ChooseInstallModDialog.cpp]: Does this resolve dependencies recursive?

File: gui/dialogs/DownloadProgressDialog.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    ui->setupUi(this);
+    ui->progressBar->setMaximum(m_modcount);
+}
+
+DownloadProgressDialog::~DownloadProgressDialog()
+{
+    delete ui;
+}
+
+void DownloadProgressDialog::ModAdded(QuickMod *mod)
```
Comment (02JanDal) [inline gui/dialogs/DownloadProgressDialog.cpp]: Please rename to modAdded (to keep to the naming scheme)

File: quickmod/QuickModDependancyResolver.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+#include <QList>
+#include "QuickModFile.h"
+#include "logic/BaseInstance.h"
+
+class QuickModDependancyResolver : public QObject
+{
+    Q_OBJECT
+public:
+    explicit QuickModDependancyResolver(QObject *parent = 0);
+    QList<QuickModFile> Resolve(BaseInstance* instance, QList<QuickModFile> files);
```
Comment (02JanDal) [inline quickmod/QuickModDependancyResolver.h]: Again, camelCase

**Comment by 02JanDal** [comment]
Apart from the code comments I wrote, either of us has messed up the formatting (probably tabs/spaces).

Also, I would prefer to have working code (filing the resolve function).

Also, as far as I can see dependencies are currently only resolved for one level.

Looks good otherwise.

I'm gonna be on irc a bit later tomorrow (about 19:00 CET).

**Comment by Khitiara** [comment]
I fixed some stuff. Should be good now.

File: gui/dialogs/ChooseInstallModDialog.cpp [hunk 5]
```diff
@@ -9,166 +9,198 @@
...
+    {
+        return;
+    }
+
+    auto choosenMod = m_view->selectionModel()->selectedRows().first().data(QuickModsList::QuickModRole).value<QuickMod*>();
+    QList<QuickMod*> modsList;
+
+    modsList.append(choosenMod);
+    dialog = new DownloadProgressDialog(choosenMod->dependentUrls().count(), this);
+    connect(&*MMC->quickmodslist(), SIGNAL(modAdded(QuickMod*)), dialog, SLOT(ModAdded(QuickMod*)));
```
Comment (02JanDal) [inline gui/dialogs/ChooseInstallModDialog.cpp]: Forgot to rename ModAdded -> modAdded?

**Comment by 02JanDal** [comment]
It also still has the formatting issue.

**Pull request merged**